src/huggingface_hub/HfRepostiory.py

```diff
@@ -0,0 +1,138 @@
+
+        self.email = email
+        self.user = user
+        self.large_file_tracking_list = [
+            "*.bin.*",
+            "*.lfs.*",
+            "*.bin",
+            "*.h5",
+            "*.tflite",
+            "*.tar.gz",
+            "*.ot",
+            "*.onnx",
+            "*.pt",
+        ] + large_file_tracking_list
+
+        self.install_lfs_in_repo()
+
+        self.clone_remote_repository_to_local_dir()
+
+        self.add_tracking_for_large_files()
+
+        # self.config_git_username_and_email()
+
+    def install_lfs_in_repo(self):
+        """installs git lfs for current shell user"""
+        subprocess.run("git lfs install".split(), check=True)
+
+    def config_git_username_and_email(self):
+        """sets git user and email for commiting to repository"""
+        subprocess.run(f"git config --global user.email {self.email}".split(), check=True)
+        subprocess.run(f"git config --global user.name {self.user}".split(), check=True)
+
+    def clone_remote_repository_to_local_dir(self):
+        """clones repository from HuggingFace Hub into model directory if directory exists init and add remote repo"""
+        # TODO: refactor exception handling to avoidtry:
+        #   try:
+        #     output = subprocess.check_output(['git', 'commit', '-m', 'commit model from SageMaker'])
+        #   except subprocess.CalledProcessError as e:
+        #       print(e.output.decode())
+        try:
+            subprocess.run(f"git clone {self.repo_url} {self.model_dir}".split(), check=True)
+        except Exception:
+            subprocess.run(f"git init .".split(), check=True, cwd=self.model_dir)
+            subprocess.run(f"git remote add origin {self.repo_url}".split(), check=True, cwd=self.model_dir)
+            subprocess.run(f"git checkout -b main".split(), check=True, cwd=self.model_dir)
+
+    def add_token_to_repository_url(self, repo_url: str, huggingface_token: str) -> str:
+        """replaces default repository url and adds huggingface token to so git push is possible"""
+        return repo_url.replace("https://", f"https://user:{huggingface_token}@")
+
+    def add_tracking_for_large_files(self):
+        """adds all large files to .gitattributes for tracking"""
+        for file_ending in self.large_file_tracking_list:
+            subprocess.run(f"git lfs track {file_ending}".split(), check=True, cwd=self.model_dir)
+
+    def create_model_card(self, dataset: str, model_id: str, hyperparameters: Dict, eval_results: List):
+        """creates a model card from an existing template needs datasets, model_id, hyperparameters and eval_results"""
+        model_card = MODEL_CARD_TEMPLATE.format(
+            dataset=dataset,
+            model_id=model_id,
+            hyperparameters=hyperparameters,
+            eval_table="\n".join(f"| {k} | {v} |" for k, v in eval_results.items()),
+        )
+        with open(os.path.join(self.model_dir, "README.md"), "w") as f:
+            f.write(model_card)
```

Comment:
in first version I don't think we should do this (here)

Reply:
removed for now